modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js

```diff
@@ -111,6 +111,11 @@ export function handleAction(
 
 		window.open(url);
 	}
+	else if (target === 'self') {
+		event.preventDefault();
+
+		window.open(url, '_self');
```

Comment:
This seems fine on first glance, as it consistently extends existing API. But, I have to say, I really dislike the existing API:
1. `target` is a very poor prop, as it is an counter-intuitive kitchen-sink of config options. 
2. It is misleading that we are mixing `target` prop with `target` attribute of `a` / `window.open`.
3. On line 112 we may have a [reverse tabnapping vulnerability](https://owasp.org/www-community/attacks/Reverse_Tabnabbing) :grimacing: 

In what case do we need `window.open`, with `_self` target? To navigate with JS, we should use `navigate()`, like for example [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/asset/asset-categories-admin-web/src/main/resources/META-INF/resources/js/ItemSelectorAddCategory.js#L121). This will result in SPA navigation, a better user experience. If we really must refresh the page, we should use `window.location.href = url`.

In either case, we don't need to extend API. We can either use `href` and set a normal link, or have `navigate` in `onClick`.

Reply:
This particular case would be for using a rest endpoint for export, in which it opens a new window. Using `window.open(url)` would have worked as well, it's just that from testing it out, it would have opened a new window that starts the download and immediately closes, versus `window.open(url, '_self')` which would just start download.

I think being able to use `href` and getting to set a normal link + attributes `download` and `target="_blank"` would work the best, although I'm not sure how to do that with the `ClayDataSetActionDropdownItem` we're using? The parameters seemed limited and I thought using `target` would be fine as there's already `blank`? Just thought this would be the simplest change to make 😅

Reply:
> This particular case would be for using a rest endpoint for export, in which we'd like to open in a new window

But, `window.open(url, '_self')`, that we are adding in this PR, does not open in a new window. It opens page in the same window, just like normal click on a link. It's the default behavior.

Sidenote - IMO JS opening something a new window is a pretty bad user experience, no matter the context. But, that's ultimately up to your product folks.

Reply:
Ah, my bad for wording.

We just would like to download a file from this export link.
<img width="1117" alt="Screen Shot 2021-12-10 at 10 47 55 AM" src="https://user-images.githubusercontent.com/14063502/145626563-b583f4b7-4c91-45f7-ad3f-50130ad740db.png">

From using the "blank" option, it performs like this. It just opens a new tab that immediately closes.

https://user-images.githubusercontent.com/14063502/145626835-de2c20b9-7a59-4d49-86a8-98aaf3ec6dff.mp4

Using this "self", it performs like this.

https://user-images.githubusercontent.com/14063502/145627447-59827418-3b6c-47c0-b521-65100d85cd93.mp4

I hope that makes it less confusing, please let us know what ideas you might have for downloading the document from the link?

Reply:
@markocikos Yes, we really like the idea of the added support to specify action items!

I added the commit to clarify the option as "download," and we will keep in mind that it's a temporary solution 👍